packages/base/src/helpers/plugin.ts

```diff
@@ -130,6 +166,65 @@ export const installPlugin = async (packageOrScope: string): Promise<boolean> =>
   }
 }
 
+interface InstallPackageOptions {
+  silent: boolean
+  dev: boolean
+}
+
+interface InstallPackageOutput {
+  exitCode: number
+  stdout: string
+  stderr: string
+}
+
+type InstallPackageFn = (names: string[], options: InstallPackageOptions) => Promise<InstallPackageOutput>
+
+// Wrapper function to be mocked in tests
+export const importInstallPkg = async () => {
+  return import('@antfu/install-pkg') as Promise<{installPackage: InstallPackageFn}>
+}
+
+const temporarilyInstallPluginWithNpx = async (scope: string) => {
+  const {basePackage, pluginPackage} = getPackagesToInstall(scope)
+
+  const emitPath = process.platform === 'win32' ? 'set PATH' : 'printenv PATH'
+  const cmd = `npx -y -p ${basePackage} -p ${pluginPackage} ${emitPath}`
+
+  debug('Using npx to install the missing plugin:', cmd)
+  const output = await new Promise<string>((resolve, reject) => {
+    exec(cmd, (error, stdout) => {
+      if (error) {
+        reject(error)
+      } else {
+        resolve(stdout)
+      }
+    })
+  })
+  debug('Output:', output)
+
+  const isWindows = process.platform === 'win32'
```

Comment:
Nit: you check this condition on line 130 too, maybe it's cleaner to move this to the beginning of the method and use it there